root/user/UserCollections.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import some from 'lodash/some';
+import * as React from 'react';
+
+import Table from '../components/Table';
+import UserAccountLayout from '../components/UserAccountLayout';
+import {withCatalystContext} from '../context';
+import {unwrapNl} from '../static/scripts/common/i18n';
+import {formatPluralEntityTypeName}
+  from '../static/scripts/common/utility/formatEntityTypeName';
+import {
+  defineNameColumn,
+  typeColumn,
+  subscriptionColumn,
+  defineActionsColumn,
+} from '../utility/tableColumns';
+
+type Props = {|
+  +$c: {user: EditorT} & CatalystContextT,
+  +collaborativeCollections: CollectionListT,
+  +ownCollections: CollectionListT,
+  +user: EditorT,
+|};
+
+type CollectionListT = {|
+  +[string]: $ReadOnlyArray<CollectionT>,
+|};
+
+const collectionsListTitle = {
+  area: N_l('Area Collections'),
+  artist: N_l('Artist Collections'),
+  event: N_l('Event Collections'),
+  instrument: N_l('Instrument Collections'),
+  label: N_l('Label Collections'),
+  place: N_l('Place Collections'),
+  recording: N_l('Recording Collections'),
+  release: N_l('Release Collections'),
+  release_group: N_l('Release Group Collections'),
+  series: N_lp('Series Collections', 'plural'),
+  work: N_l('Work Collections'),
+};
+
+function formatCollaboratorNumber(
+  collaborators: $ReadOnlyArray<EditorT>,
+  currentUser: EditorT | null,
+) {
+  const isCollaborator = !!currentUser && some(
+    collaborators,
+    (collaborator => collaborator.id === currentUser.id),
+  );
+
+  return isCollaborator ? (
+    exp.l('{collaborator_number} (including you)',
+          {collaborator_number: collaborators.length})
+  ) : (
+    collaborators.length
+  );
+}
+
+function formatPrivacy(isPublic: boolean) {
+  return isPublic ? l('Public') : l('Private');
+}
+
+function collaborativePrivacy(
+  collection: CollectionT,
+  currentUser: EditorT | null,
+) {
+  return !!currentUser && !!collection.editor &&
+    collection.editor.id === currentUser.id ? (
+      formatPrivacy(collection.public) + ' ' + l('(your collection)')
+    ) : (
+      formatPrivacy(collection.public)
+    );
+}
+
+const CollectionsEntitySection = ({
```

Comment:
Why not just CollectionsTable or, if you don't like Table because it includes the heading, CollectionsList or even CollectionsSection? "Entity" just seems weird to me here.

Reply:
This component is a _section_ (that features not only a table but also an `h3` title) and matches a specific _entity_ type (not the same as own/collaborative sections). Would `CollectionsEntityTypeSection` be more explicit?

(Note: This is just a 1:1 conversion from TT to JSX but these sections could possibly be merged in the future as tables get the client-sided ability of filtering by entity type.)

Reply:
Fixed in https://github.com/metabrainz/musicbrainz-server/pull/1246/commits/6ee5d3d424b1dda8aecefe6be1b9f7d2ac4092e1#diff-3f513fa6fa1fed40110c8c20d602c01fR79